### DISCUSSION TITLE
"Property Email was not yet loaded" when i try to get the Email of a SharePoint people/group field

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [pnp/pnpcore#1404] [Q&A]
I am using LoadListDataAsStreamAsync to get the SharePoint list items as follow:-

```
            string viewXml = @"<View Scope='RecursiveAll'>
                    <ViewFields>
                      <FieldRef Name='Title' />
                      <FieldRef Name='Created' />
                      <FieldRef Name='FileLeafRef' />
                        <FieldRef Name='WorkOrderStatus' />
                        <FieldRef Name='WorkOrderType' />
                        <FieldRef Name='TechnicianName' />
                        <FieldRef Name='FileLeafRef' />
                        <FieldRef Name='AssetID' />
                    </ViewFields>
                    <Query>
<Eq>
        <FieldRef Name='FSObjType' />
        <Value Type='Integer'>0</Value>
    </Eq>
                    </Query>
                   </View>";

            // Load all the needed data using paged requests
            bool paging = true;
            string nextPage = null;
            while (paging)
            {
                var output = await workOrderList.LoadListDataAsStreamAsync(new RenderListDataOptions()
                {
                    ViewXml = viewXml,
                    RenderOptions = RenderListDataOptionsFlags.ListData,
                    Paging = nextPage ?? null,
                }).ConfigureAwait(false);

                if (output.ContainsKey("NextHref"))
                {
                    nextPage = output["NextHref"].ToString().Substring(1);
                }
                else
                {
                    paging = false;
                }
            }

            // Iterate over the retrieved list items
            foreach (var listItem in workOrderList.Items.AsRequested())
            {
                // Do something with the list item
                Console.WriteLine(listItem["AssetID"]);
            }
```

but when i try to get the email of the TechnicianName field as follow :-

 ```
var groupedWorkOrders = workOrderList.Items.AsRequested()
.Select(wo =>
{
    var technicianName = wo["TechnicianName"] != null ? (wo["TechnicianName"] as FieldUserValue).Email.ToString() : string.Empty;
```

i got this error "Property Email was not yet loaded" .. so how i can load the email property inside my CAML?

**Comment by jansenbe** [comment]
@mvcsharepointdev : when I try this using LoadListDataAsStreamAsync I do get the required person details. Are you sure `TechnicianName` is a Person field? 

![image](https://github.com/pnp/pnpcore/assets/7451219/949fef67-11da-4c3f-80ea-95b48cbc8410)

**Reply by mvcsharepointdev** [reply]
@jansenbe yes it is a Person/group field, as follow:-

![image](https://github.com/pnp/pnpcore/assets/79726511/10fdec8e-b3ab-4760-8599-58fa17f00135)


are you limiting the fields you are getting from SharePoint (as i am doing) ? or you are getting all list items fields?

**Reply by jansenbe** [reply]
I've a CAML query set that only loads a couple of fields. What you can do is use Fiddler and inspect if the data you need is passed over or not

**Reply by jansenbe** [reply]
Also, if a SharePoint Group was added instead of a person then there would not be an email address...

**Reply by mvcsharepointdev** [reply]
@jansenbe this only allow users

**Reply by mvcsharepointdev** [reply]
@jansenbe i am using .net core 8.0 console application... which version you are using? and i have "options.PnPContext.GraphFirst = true"

**Reply by jansenbe** [reply]
I'm testing with .NET 8, the GraphFirst setting is unrelated here as this is a SharePoint REST request under the covers.
